c_src/ep_decoder.c

```diff
@@ -113,12 +113,11 @@ unpack_int64_as_bin(ErlNifEnv *env, ep_dec_t *dec, ERL_NIF_TERM *term)
         val |= ((uint64_t) (*(dec->p) & 0x7f) << shift);
         if ((*(dec->p)++ & 0x80) == 0) {
             size_t size = 0;
-            do {
+            while (val > 0) {
```

Comment:
the old code would have written 0 for when val is 0.

Reply:
i did not want that to happen.
reason - for clients when sending the binary - default value = 0, so this will not be in the binary and we will decode it to be empty binary... so client's 0 is the same as our empty binary or undefined.
say for example we somehow set uid value to be empty binary and then encode it.. then after decoding it - we still want empty binary and not <<"0">>.. so updated it to avoid that corner case.
for undefined - it would work as expected - decoding it to be empty binary.